Management/Clients/CovidDataClient.cs

```diff
@@ -1,28 +1,131 @@
 ﻿using System;
+using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
+using Management.Clients.Models;
 using Management.DomainModels;
 using Management.Enum;
 using Management.Interface;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using RestSharp;
 
 namespace Management.Clients
 {
     /// <summary>
-    /// Representation of the client for the COVID-19 data API.
+    /// Representation of the client for the covid API.
     /// </summary>
     public class CovidDataClient : ICovidDataClient
     {
+        private readonly IConfiguration _configuration;
+
+        private IRestClient _restClient;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CovidDataClient"/> class.
+        /// </summary>
+        /// <param name="configuration">configuration for weather data client.</param>
+        public CovidDataClient(IConfiguration configuration)
+        {
+            _restClient = new RestClient("https://api.covidtracking.com");
+            _configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+        }
+
+        /// <summary>
+        /// Retrieve city data from covid api
+        /// </summary>
+        /// <param name="state">state of interest eg. NY.</param>
+        /// <param name="cancellationToken">used to signal that the asynchronous task should cancel itself.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation, with a status code.</returns>
+        public async Task<CovidStateResponse> GetStateCovidDataAsync(
+            State state,
+            CancellationToken cancellationToken)
+        {
+            var request = new RestRequest("v1/states/" + state.Value.ToLower() + "/current.json", Method.GET);
+
+            var response = await _restClient.ExecuteAsync(request, cancellationToken);
+
+            if (response.StatusCode == System.Net.HttpStatusCode.OK)
+            {
+                return JsonConvert.DeserializeObject<CovidStateResponse>(response.Content);
+            }
+            else if (response.StatusCode == System.Net.HttpStatusCode.Forbidden)
+            {
+                return new CovidStateResponse
+                {
+                   Positive = new Random().Next(0, 99999999),
```

Comment:
is 99999999 too much? in terms of fake the positive case in a State.

Reply:
Same here, did you forget to push your commit

Reply:
Removed because found out that the current covid API doesn't have limits.